src/subgradients.jl

```diff
@@ -0,0 +1,34 @@
+
+@doc raw"""
+    subgrad_distance(M, q, p[, atol=eps(eltype(p))])
+    subgrad_distance!(M, X, q, p[, atol=eps(eltype(p))])
+
+compute the subgradient of the distance (in place of `X`)
+
+```math
+f(p) = d_{\mathcal M}(p, q)
+```
+
+to a fixed point `q` on the manifold `M`.
+The subgradient at `p = q`is given by a norm one tangent vector at `p`.
+
+# Optional
+
+* `atol` – (`eps(eltype(p))`) the tolerance to use when evaluating the distance between `p` and `q`.
+"""
+function subgrad_distance(M, q, p; atol = eps(eltype(p)))
+    @assert distance(M, q, p) ≤ atol ["Since p and q are different, use grad_distance."]
```

Comment:
No, I would not do this, it is a valid case here as well, no need for a warning.